src/Feature/Editor/SurfaceView.re

```diff
@@ -73,24 +81,74 @@ let%component make =
 
   let {scrollX, scrollY, _}: Editor.t = editor;
 
-  let onMouseUp = (evt: NodeEvents.mouseButtonEventParams) => {
-    Log.trace("editorMouseUp");
-
+  let getMaybeLocationFromMousePosition = (mouseX, mouseY) => {
     maybeBbox^
-    |> Option.iter(bbox => {
+    |> Option.map(bbox => {
          let (minX, minY, _, _) = bbox |> BoundingBox2d.getBounds;
 
-         let relY = evt.mouseY -. minY;
-         let relX = evt.mouseX -. minX;
+         let relX = mouseX -. minX;
+         let relY = mouseY -. minY;
 
-         let (line, col) =
-           Editor.Slow.pixelPositionToBufferLineByte(
-             ~buffer,
-             ~pixelX=relX,
-             ~pixelY=relY,
-             editor,
-           );
+         Editor.Slow.pixelPositionToBufferLineByte(
+           ~buffer,
+           ~pixelX=relX,
+           ~pixelY=relY,
+           editor,
+         );
+       });
+  };
+
+  let onMouseMove = (evt: NodeEvents.mouseMoveEventParams) => {
+    getMaybeLocationFromMousePosition(evt.mouseX, evt.mouseY)
+    |> Option.iter(((line, col)) => {
+         dispatch(
+           Msg.MouseMoved({
+             location:
+               EditorCoreTypes.Location.create(
+                 Index.fromZeroBased(line),
+                 Index.fromZeroBased(col),
+               ),
+           }),
+         )
+       });
+    hoverTimerActive := true;
+    lastMousePosition := Some((evt.mouseX, evt.mouseY));
+    resetHoverTimer();
+  };
+
+  let%hook () =
+    Hooks.effect(
+      If(
+        (t, _) => Revery.Time.toFloatSeconds(t) > Constants.hoverTime,
```

Comment:
This seems a bit unclear to me - why are we ignoring the second argument for it?

Reply:
Ah yeah I should probably add a comment about that. Usually the `If` hook compares `x` to a previous version of itself. However I want to compare it to a constant value. Theoretically I can just move it to an isolated if statement, but I think the hook provides clarity that it's an effect and also runtime benefits, if I'm not mistaken.

Reply:
Makes sense to keep it an effect, thanks for the clarification! A comment would be helpful 👍